compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -196,7 +196,9 @@ impl<'hir> Map<'hir> {
                 ItemKind::Macro(_, macro_kind) => DefKind::Macro(macro_kind),
                 ItemKind::Mod(..) => DefKind::Mod,
                 ItemKind::OpaqueTy(..) => DefKind::OpaqueTy,
-                ItemKind::TyAlias(..) => DefKind::TyAlias,
+                ItemKind::TyAlias(..) => {
+                    DefKind::TyAlias { lazy: self.tcx.features().lazy_type_alias }
```

Comment:
hmm... I wonder why we duplicate this logic with compiler/rustc_resolve/src/build_reduced_graph.rs ... Not the fault of this PR, but something we should investigate